### PR TITLE
added error handling for saving with zero recipients selected

### PR DESCRIPTION
**Pull request by KyleCardwell**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

- adds error handling to the Contact List page to prevent users from saving while zero contacts are selected (at least 1 must be selected to enable saving)
- if a user attempts to save with zero contacts selected, an error will appear and the first checkbox on the page will be focused.

### [MHV-59350](https://jira.devops.va.gov/browse/MHV-59350) - Implement error handling for Contact list functionality###

> Implement error handling for Contact list functionality
> 
> Complete/latest requirements located in Figma:  https://www.figma.com/design/uddjxccqvUyz5FSg1aD5wi/ZFeat-{}Triage-Group-Selection-Flows?node-id=792-21547&t=1UMB6N8tImODPlq8{-}
> 
> User Story: As a Secure Messaging User, I want to be able to add or edit my contact list and understand any errors I may receive.  
> 
> Given: The user is unable to locate the needed group in the dropdown list in the "To" field
> When: The user can click on the link, "If you can't find your team" and get more information and help to navigate the problem
> Then:  The user can navigate to the Contact list page and add/edit the groups the user is associate with, save changes and see a success banner.
> 
> When:  the user encounters an error while editing the Contact list
> 
> Then:  the user sees the error message instructing the user how to fix the problem
> 
> Feature Flag Y/N ? Y-For all contact list functionality
> Manual Testing Y/N ? Y
> Automated Testing Y/N ? Y-Fazil
> Accessibility Testing Y/N ? Y
> 
> UCD Validation Y/N ? Y
> 
> Acceptance Criteria:
> AC1  User sees error content when an error occurs while saving the contact list changes
> AC2  UCD validation
> 
> Links to UCD: 
> FIGMA Link: https://www.figma.com/design/uddjxccqvUyz5FSg1aD5wi/ZFeat---Triage-Group-Selection-Flows?node-id=792-21547&t=1UMB6N8tImODPlq8-4

## Testing done

- updated unit tests
- manual testing to validate that a user cannot save when there are no contacts selected
- accessibility testing to check what errors are announced

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile - one facility |        |Screenshot 2024-08-29 at 11 03 00 AM|
| Desktop - one facility |        |Screenshot 2024-08-29 at 11 01 15 AM|
| Mobile - multiple facilities |        |Screenshot 2024-08-29 at 11 03 32 AM|
| Desktop - multiple facilities |        |Screenshot 2024-08-29 at 11 02 11 AM|

## What areas of the site does it impact?

Va.gov - Secure Messaging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/containers/EditContactList.jsx [hunk 1]
```diff
@@ -81,23 +88,26 @@ const EditContactList = () => {
 
   const handleSaveAndExit = async (e, forceSave) => {
     e.preventDefault();
+    if (!isMinimumSelected) {
+      setMinimumSelectError(ErrorMessages.ContactList.MINIMUM_SELECTION);
+      focusOnErrorField();
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/containers/EditContactList.jsx:93]: as we discussed, this focus handler is redundant with the one in `useEffect` on line 160. I suggest either 
- adding `await` to `setMinimumSelectError` on line above and remove `useEffect` on line 157 or
- remove `focusOnErrorField` on line 93 and keep `useEffect`

File: src/applications/mhv-secure-messaging/util/testUtils.js [hunk 2]
```diff
@@ -60,3 +60,13 @@ export const checkVaCheckbox = (checkboxGroup, bool) => {
...
+
+export const getProps = element => {
+  let prop;
+  Object.keys(element).forEach(key => {
+    if (key.match(/^__react[^$]*(\$.+)$/)) {
+      prop = key;
+    }
+  });
+  return prop;
+};
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/util/testUtils.js:72]: thank you for moving this to utils 👍🏻

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]

File: src/applications/mhv-secure-messaging/tests/e2e/pages/ContactListPage.js [hunk 3]
```diff
@@ -79,6 +93,12 @@ class ContactListPage {
   clickBackToInbox = () => {
     cy.get(Locators.BACK_TO).click();
   };
+
+  verifyEmptyContactListAlert = () => {
+    cy.get(`#checkbox-error-message`)
+      .should(`be.visible`)
+      .and(`contain.text`, Alerts.CONTACT_LIST.EMPTY);
+  };
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/pages/ContactListPage.js:101]: I think we may want to add validating that correct checkbox receives proper focus after clicking `Save` and all checkboxes are blank.

File: src/applications/mhv-secure-messaging/tests/e2e/secure-messaging-contact-list-interface.cypress.spec.js [hunk 4]
```diff
@@ -13,40 +13,46 @@ describe('Secure Messaging Contact list', () => {
...
     ContactListPage.clickSaveAndExitButton();
     ContactListPage.verifyContactListSavedAlert();
 
+    cy.wait('@savedList')
+      .its('request.body')
+      .then(req => {
+        const obj = req.updatedTriageTeams.find(el => el.name.includes(`100`));
+        expect(obj.preferredTeam).to.eq(true);
+      });
+
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/secure-messaging-contact-list-interface.cypress.spec.js:50]: I would suggest adding a coverage for a scenario with 2 or more facilities

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
@KyleCardwell thank you for clearing up that redundant focus handler. 

@fazilqa good coverage. Couple things I think we can contribute to improve test coverage. Please see my comments below

File: src/applications/mhv-secure-messaging/tests/e2e/secure-messaging-contact-list-interface.cypress.spec.js (hunk 4 above)
Comment (fazilqa) [reply src/applications/mhv-secure-messaging/tests/e2e/secure-messaging-contact-list-interface.cypress.spec.js:61]: done

File: src/applications/mhv-secure-messaging/tests/e2e/pages/ContactListPage.js (hunk 3 above)
Comment (fazilqa) [reply src/applications/mhv-secure-messaging/tests/e2e/pages/ContactListPage.js:104]: done

**Review by alex-morgun-1 — DISMISSED** [review]
@fazilqa thank you for including other testing scenarios and additional validations ✅

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**